cmd/server/main.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"net/http"
+	"os"
+	"path"
+
+	appLogger "github.com/ShukinDmitriy/GophKeeper/internal/logger"
+
+	_ "github.com/ShukinDmitriy/GophKeeper/cmd/server/docs"
+	"github.com/ShukinDmitriy/GophKeeper/internal/server"
+	"github.com/ShukinDmitriy/GophKeeper/internal/server/auth"
+	"github.com/ShukinDmitriy/GophKeeper/internal/server/config"
+	"github.com/ShukinDmitriy/GophKeeper/internal/server/controllers"
+	"github.com/ShukinDmitriy/GophKeeper/internal/server/repositories"
+	"github.com/golang-migrate/migrate/v4"
+	migratePostgres "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/gommon/log"
+	"go.uber.org/fx"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+// main Запуск сервера
+// @title Swagger EOL API
+// @version 1.0
+// @BasePath /api
+func main() {
+	fx.New(
+		fx.Provide(
+			// Конфигурация
+			func() *config.Config {
+				conf, err := config.NewConfig()
+				if err != nil {
+					panic(err)
+				}
+				return conf
+			},
+			// логгер
+			func(conf *config.Config) appLogger.Logger {
+				return appLogger.NewLogger(conf.LogLevel, conf.LogPath)
+			},
+			// база данных
+			func(
+				conf *config.Config,
+				appLog appLogger.Logger,
+			) *gorm.DB {
+				postgresqlURL := conf.DatabaseURI
+
+				if postgresqlURL == "" {
+					appLog.Fatal("no DATABASE_URI in server.env")
+					return nil
+				}
+
+				var level logger.LogLevel
+				switch conf.LogLevel {
+				case log.DEBUG:
+				case log.INFO:
+					level = logger.Info
+				case log.WARN:
+					level = logger.Warn
+				case log.ERROR:
+					level = logger.Error
+				case log.OFF:
+					level = logger.Silent
+				}
+
+				db, err := gorm.Open(postgres.Open(postgresqlURL), &gorm.Config{
+					Logger: logger.Default.LogMode(level),
+				})
+				if err != nil {
+					appLog.Fatal(err)
+					return nil
+				}
+
+				return db
+			},
+			// Репозитории:
+			// пользователя
+			func(DB *gorm.DB) *repositories.UserRepository {
+				return repositories.NewUserRepository(DB)
+			},
```

Comment:
```suggestion
      repositories.NewUserRepository,
```

у тебя есть привычка все усложнять

Reply:
заменил, где был просто return